bundle/bin/rke2-uninstall.ps1

```diff
@@ -292,10 +293,23 @@ function Remove-Containerd () {
             foreach ($image in $images) {
                 Remove-Image $ns $image
             }
-            Remove-Namespace $ns
-            # TODO
-            # clean pods with crictl
-            # $CONTAINER_RUNTIME_ENDPOINT = "npipe:\\.\\pipe\\containerd-containerd"
+        }
+
+        # Some resources in the namespace take a while to disappear. Try several times to remove the namespace and give up after 30s 
```

Comment:
Why is it so critical that this be removed? There's only ever the one namespace, `k8s.io` - and it can't be removed until everything in it is gone. Deleting the namespace doesn't trigger deletion of the images and containers inside it, those need to be done manually but I don't see that we're doing that here. Do we need to retry removing the specific resources that are blocking its deletion, instead of just retrying the namespace delete and hoping the other things have gone away in the background?

Reply:
Probably my comment confused you. In the previous lines, we are removing tasks, containers and images, which should remove all resources. The problem is that by the time we start removing namespaces, some of those resources are still available, probably set for deletion, but available. That makes the deletion of the namespace fail. Then, we move on and uninstall containerd, rke2, kube-proxy... We are killing everything when those resources are still available and that ends up badly. We end up with containerd files (normally related to snapshots) remiaining in the file system and impossible to remove

Reply:
While being at the gym, I thought that perhaps we could try removing the 30s. countdown and just verify everything (tasks, containers, images and snapshots) is empty before removing the namespace. However, if for example, the user forgot to drain, containerd will never be empty, so we would get stuck forever. That's why I think we need some kind of upper limit

Reply:
> if for example, the user forgot to drain, containerd will never be empty

Is that because the kubelet will still be running, trying to restart the containers?

It sounds like we need to kill the kubelet to prevent it from restarting things, but leave containerd running so we can use it to kill containers by hand. Is that correct? This would probably require some changes to the RKE2 golang code, to allow for a partial node shutdown that allows for stopping the kubelet but not containerd. I don't think that asking the user to drain the node is going to work - even with a drain, some pods remain running. It is also asking a lot of the user to have them drain before running killall. Killall should be able to clean things up even if the node wasn't prepared properly.

Reply:
I changed the behaviour a bit. Now the rke2-uninstall.ps1 creates a lock file in DataDir and then kills kubelet. If kubelet dies, pebinary checks if that lock file exists and if it does, it does not restart kubelet. The lock file gets removed when the script removes the DataDir, so it won't affect new rke2 installations